### PR TITLE
first proposal draft

### PR DESCRIPTION
**Pull request by IAmMorrow**

### CODE REVIEW

File: packages/simulator/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "@tsconfig/node16-strictest-esm/tsconfig.json",
   "compilerOptions": {
+    "module": "commonjs",
+    "esModuleInterop": true,
```
Comment (Justkant) [inline packages/simulator/tsconfig.json:5]: Why do we need this changes ?

**Review by Justkant — COMMENTED** [review]
Some questions but I think we can go forward with this

File: packages/server/src/errors.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+export const ACCOUNT_NOT_FOUND: JSONRPC.RpcResponseError = {
+  code: JSONRPC.RpcErrorCode.SERVER_ERROR,
+  message: "The specified account was not found in this wallet",
+};
+
+export const CURRENCY_NOT_FOUND: JSONRPC.RpcResponseError = {
+  code: JSONRPC.RpcErrorCode.SERVER_ERROR,
+  message: "The specified currency was not found in this wallet",
```
[the remark this answers is not in the document]
Comment (IAmMorrow) [reply packages/server/src/errors.ts:15]: Errors messages are still placeholders for now. But only the error code itself is important for error handling in implementations

File: packages/server/src/index.ts [hunk 3]
```diff
@@ -1,94 +1,101 @@
...
+
+  private async handleMessage(payload: unknown) {
+    // TODO validate JSONRPC request
+    const rpcRequest = payload as JSONRPC.RpcRequest;
+    if (rpcRequest.method in internalHandlers) {
+      const methodId = rpcRequest.method as keyof typeof internalHandlers;
+      try {
+        const internalHandler = internalHandlers[methodId];
+
+        // await this.runMiddlewares();
```
[the remark this answers is not in the document]
Comment (IAmMorrow) [reply packages/server/src/index.ts:37]: Yup, should not be too hard and can be added later

File: packages/simulator/tsconfig.json (hunk 1 above)
Comment (IAmMorrow) [reply packages/simulator/tsconfig.json:5]: Was necessary to "execute" a ts file with ts-node.

File: packages/server/src/index.ts [hunk 4]
```diff
@@ -1,94 +1,101 @@
...
     }
+  }
 
-    this.transport.onMessage = (payload) =>
-      serverAndClient.receiveAndSend(payload, { logger: this.logger });
-    this.transport.connect();
-    this.serverAndClient = serverAndClient;
-    this.logger.log("connected", this.transport);
+  setCurrencies(currencies: Currency[]) {
+    this.walletContext.currencies$.next(currencies);
```
[the remark this answers is not in the document]
Comment (IAmMorrow) [reply packages/server/src/index.ts:73]: Yup that's ongoing work

File: packages/server/src/errors.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Justkant) [reply packages/server/src/errors.ts:15]: I agree that the error code is enough to develop but the idea is to help when debugging by adding as much context as possible in the message.

**Review by Justkant — APPROVED** [review]

**Pull request merged**